vizro-core/docs/pages/user-guides/dynamic-data.md

```diff
@@ -0,0 +1,103 @@
+# Dynamic data
+
+A dynamic data source is a Python function that returns a pandas DataFrame. This function is executed when the dashboard is initially started and _can be executed again while the dashboard is running_. This makes it possible to refresh the data shown in your dashboard without restarting the dashboard itself. If you do not this functionality then you should use [static data](static-data.md) instead.
+
+Unlike static data, dynamic data cannot be supplied directly into the `data_frame` argument of a `figure`. Instead, it must first be added to the Data Manager and then referred to by name.
+
+!!! example "Dynamic data"
```

Comment:
Just as a quick thing for now I will do something where it just plots the iris dataset but with different random points selected or something like that. And then in the future we can make this a more interesting example like the weather where we call some external API.

Reply:
Done in [acb546a](https://github.com/mckinsey/vizro/pull/398/commits/acb546a8a76658ab09b1c9e1b6c3b7559dfc694e) - see what you think.

@maxschulz-COL you don't know of any reason why we wouldn't be able to host `iris.csv` on RTD and download it as done here?

Reply:
No, but I also haven't tried it. But I like the new approach. The only thing I am unsure about is the following:

Should the code example really have `pd.read_csv` (I know it is to make it clear you are fetching from a file) or actually `px.data.iris()`, ie something that will always work?

I kinda prefer the former, but with a message similar to the one we already have, ie that here we kinda simulate the dynamic case

Reply:
Yeah, I had this done with `px.data.iris()` originally and then decided that wasn't clear enough because it's too far removed from a "real" working example where the data can change. So better to use `pd.read_csv` even though that means downloading a file to get the example working.

The download works ok on the automatically built docs for this PR so presumably it will also work with the released docs 🤞